File: app/src/main/java/dev/ragnarok/fenrir/db/interfaces/ITopicsStore.java
package dev.ragnarok.fenrir.db.interfaces;

import androidx.annotation.CheckResult;
import androidx.annotation.NonNull;

import java.util.List;

import dev.ragnarok.fenrir.db.model.entity.OwnerEntities;
import dev.ragnarok.fenrir.db.model.entity.PollEntity;
import dev.ragnarok.fenrir.db.model.entity.TopicEntity;
import dev.ragnarok.fenrir.model.criteria.TopicsCriteria;
import io.reactivex.rxjava3.core.Completable;
import io.reactivex.rxjava3.core.Single;


public interface ITopicsStore {

    @CheckResult
    Single<List<TopicEntity>> getByCriteria(@NonNull TopicsCriteria criteria);

    @CheckResult
    Completable store(int accountId, int ownerId, List<TopicEntity> topics, OwnerEntities owners, boolean canAddTopic, int defaultOrder, boolean clearBefore);

    @CheckResult
    Completable attachPoll(int accountId, int ownerId, int topicId, PollEntity pollDbo);
}

File: app/src/main/java/dev/ragnarok/fenrir/db/interfaces/ITopicsStore.kt
package dev.ragnarok.fenrir.db.interfaces

import androidx.annotation.CheckResult
import dev.ragnarok.fenrir.db.model.entity.OwnerEntities
import dev.ragnarok.fenrir.db.model.entity.PollEntity
import dev.ragnarok.fenrir.db.model.entity.TopicEntity
import dev.ragnarok.fenrir.model.criteria.TopicsCriteria
import io.reactivex.rxjava3.core.Completable
import io.reactivex.rxjava3.core.Single

interface ITopicsStore {
    @CheckResult
    fun getByCriteria(criteria: TopicsCriteria): Single<List<TopicEntity>>

    @CheckResult
    fun store(
        accountId: Int,
        ownerId: Int,
        topics: List<TopicEntity>,
        owners: OwnerEntities?,
        canAddTopic: Boolean,
        defaultOrder: Int,
        clearBefore: Boolean
    ): Completable

    @CheckResult
    fun attachPoll(accountId: Int, ownerId: Int, topicId: Int, pollDbo: PollEntity?): Completable
}